var/spack/repos/builtin/packages/py-yt/package.py

```diff
@@ -57,6 +64,15 @@ class PyYt(PythonPackage):
     depends_on("py-sympy@1.0:", type=('build', 'run'))
     depends_on("rockstar@yt", type=('build', 'run'), when="+rockstar")
     depends_on("python@2.7.0:2.7,3.5:", type=('build', 'run'))
+    depends_on("py-cmyt@0.2.2:", type=('build', 'run'), when="@4.0.0:")
+    depends_on("py-more-itertools@8.4.0:", type=('build', 'run'), when="@4.0.0:")
+    depends_on("py-numpy@1.13.3:", type=('build', 'run'), when="@4.0.0:")
+    depends_on("py-pyparsing", type=('build', 'run'), when="@4.0.0:")
+    depends_on("py-toml", type=('build', 'run'), when="@4.0.0:4.0.2")
+    depends_on("py-tomli@1.2.3:", type=('build', 'run'), when="@4.0.3:")
+    depends_on("py-tomli-w@0.4.0:", type=('build', 'run'), when="@4.0.3:")
+    depends_on("py-tqdm@3.4.0:", type=('build', 'run'), when="@4.0.0:")
+    depends_on("py-unyt@2.8.0:", type=('build', 'run'), when="@4.0.0:")
```

Comment:
Missing build dep on wheel 0.36.2+, need to update cython, python, matplotlib, etc. deps. If the deps have drastically changed you might find it easier to do something like:
```python
with when('@:3'):
    # old deps

with when('@4:'):
    # new deps
```